addons/lunch/models/lunch_product.py

```diff
@@ -60,6 +68,13 @@ def write(self, vals):
                 topping_values.update({'topping_category': 3})
         return super(LunchProductCategory, self).write(vals)
 
+    def toggle_active(self):
+        """ Archiving related lunch product """
+        res = super().toggle_active()
+        products = self.env['lunch.product'].with_context(active_test=False).search([('category_id', 'in', self.ids), ('active', 'not in', self.mapped('active'))])
```

Comment:
I'm afraid it's not going to work as intended.
If  you call the method on 2 categories (one archived and the other active), no products will be found.

I'd do something like here : https://github.com/odoo/odoo/blob/17e409181efa89d950443e3a5c5c7943132dfdd5/addons/coupon/models/coupon_program.py#L124-L127

Reply:
@kba-odoo , 👍  Good point. done the changes. thanks